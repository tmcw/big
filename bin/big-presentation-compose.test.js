/* eslint-env jest */
var fs = require("fs");
var os = require("os");
var path = require("path");
var compose = require("./big-presentation-compose.js");

var workingDirectory;

beforeEach(function() {
  var tmp = path.join(os.tmpdir(), "./tmp-" + Date.now());
  fs.mkdirSync(tmp);
  workingDirectory = tmp;
});

test("compose", function() {
  fs.writeFileSync(path.join(workingDirectory, "./index.md"), "# hi");
  process.chdir(workingDirectory);
  compose();
  expect(fs.readdirSync(workingDirectory)).toMatchSnapshot();
  expect(
    fs.readFileSync(path.join(workingDirectory, "index.html"), "utf8")
  ).toMatchSnapshot();
});
