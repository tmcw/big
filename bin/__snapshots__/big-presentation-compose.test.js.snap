// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`compose 1`] = `
Array [
  "big.css",
  "big.js",
  "highlight.css",
  "highlight.js",
  "index.html",
  "index.md",
  "rubik-v4-latin-regular.woff",
  "rubik-v4-latin-regular.woff2",
]
`;

exports[`compose 2`] = `
"<!DOCTYPE html>
<html>
<head>
  <meta charset='utf-8'>
  <meta name='viewport' content='width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=0' />
  <title>big</title>
  <link href='big.css' rel='stylesheet' type='text/css' />
  <link href='highlight.css' rel='stylesheet' type='text/css' />
  <style>
    .new-shiny { background: #aaaaaa; }
  </style>
  <script src='big.js'></script>
  <script src='highlight.js'></script>
  <script>hljs.initHighlightingOnLoad();</script>
</head>
<body class='light'>
  <div><h1 id=\\"hi\\">hi</h1>
</div>
</body>
</html>
"
`;
